niceday_client/niceday_client.py

```diff
@@ -1,185 +1,188 @@
+                             'stored on the Senseserver changed?')
+
+        return_profile = {}
+        for k in USER_PROFILE_KEYS:
+            if k not in user_data['userProfile']:
+                raise ValueError(f'"userProfile" dict returned from '
+                                 f'niceday-api does not contain expected '
+                                 f'key "{k}". Has the data structure '
+                                 f'stored on the Senseserver changed?')
+            return_profile[k] = user_data['userProfile'][k]
+
+        return return_profile
+
+    def post_message(self, recipient_id: int, text: str):
+        """
+        Post a message to the niceday server.
+
+        Args:
+            recipient_id: user id of the recipient
+            text: text message to send
+
+        """
+        url = self._niceday_api_uri + 'messages/'
+        body = {
+            "recipient_id": recipient_id,
+            "text": text
+        }
+        return self._call_api('POST', url, body=body)
+
+    def set_user_tracker_statuses(self, user_id: int, tracker_statuses: typing.List[TrackerStatus]):
+        """
+        Set tracker statuses for a specific user.
+
+        Example Usage:
+            ```
+            self.set_user_tracker_statuses(12345, [TrackerStatus(trackerId=Tracker.SMOKING, isEnabled=True)])
+
+        Args:
+            user_id: ID of the user we want to set tracker statuses for
+            tracker_statuses: List of TrackerStatus objects.
+        """
+        url = self._niceday_api_uri + 'usertrackers/statuses'
+        body = {
+            "userId": user_id,
+            "trackerStatuses": [ts.__dict__ for ts in tracker_statuses]
+        }
+        return self._call_api('POST', url, body=body)
+
+    def get_smoking_tracker(self, user_id: int, start_time: datetime.datetime,
+                            end_time: datetime.datetime):
+        """
+        Get smoking tracker data for specific user
+
+        Args:
+            user_id: ID of the user we want to set tracker statuses for
+            start_time: The start of the time range for which to get data
+            end_time: The end of the time range for which to get data
+
+        Returns:
+            A list of smoking tracker entries, each entry is a dict containing
+            amongst others the startTime & endTime, value (which is itself a dict
+            containing the key 'quantity' that depicts the number of cigarettes smoked
+            in an entry).
+
+        """
+        url = self._niceday_api_uri + 'usertrackers/smoking/' + str(user_id)
+        query_params = {'startTime': start_time.isoformat() + 'Z',
+                        'endTime': end_time.isoformat() + 'Z'}
+        query_response = self._call_api('GET', url, query_params=query_params)
+        # convert the json response into a list of dict
+        response_json = json.loads(query_response.content)
+        return response_json

```

Comment:
This is the only change right? Looks good to me :)

Reply:
Yes you're right. I will take a look at the issue to check why this happened.